src/postgres-util/src/lib.rs

```diff
@@ -134,3 +134,34 @@ pub async fn publication_info(
 
     Ok(table_infos)
 }
+
+pub async fn drop_replication_slots(conn: &str, slots: &[&str]) -> Result<(), anyhow::Error> {
+    let (client, connection) = tokio_postgres::connect(&conn, NoTls).await?;
+    tokio::spawn(connection);
+
+    for slot in slots {
+        let active_pid = query_pg_replication_slots(&client, slot).await?;
+        if let Some(pid) = active_pid {
+            client
+                .query("SELECT pg_terminate_backend($1)", &[&pid])
+                .await?;
+        }
```

Comment:
Oh, shoot. I suspect calling `pg_terminate_backend` will be problematic in a lot of real world deployments. That requires a superuser account, and you probably don't want to give Materialize that kind of access to your Postgres instance. Ugh. We'll probably need to sort out some of way of forcing the worker threads to drop their connections before `drop_replication_slots` runs.

Reply:
If we drop the replication slot through a replication connection using the `DROP_REPLICATION_SLOT` command we can use the `WAIT` modifier there that will wait for all users to stop.

That plus a reasonable timeout should do the trick with our existing permissions, how does it sound?